userena/managers.py

```diff
@@ -225,9 +225,15 @@ def delete_expired_users(self):
         """
         deleted_users = []
         for user in get_user_model().objects.filter(is_staff=False, is_active=False):
```

Comment:
Rather than iterating over all users and then checking with hasattr, I think that it would be better to iterate over the signups so that you only get the users managed by userena, eg:

```
for signup in UserenaSignup.objects.select_related("user").filter(user__is_staff=False, user__is_active=False):
```

Then the implementation on L230 only needs to change to `signup.user.delete()`.

The exception you've added on L233 is too broad and should be removed. It might be better to add your own management command in your app to handle the constraints that you have on your models.

Reply:
importing UserenaSignup gives me the error: 
"ImportError: cannot import name 'UserenaSignup' from partially initialized module 'userena.models'   (most likely due to a circular import)"

This i can fix by importing UserenaSignup in the function, which is much less propper i think.


The database error is from (django 2.2.17) umessages:
MySQLdb._exceptions.IntegrityError: (1451, 'Cannot delete or update a parent row: a foreign key constraint fails (`eiland`.`umessages_message`, CONSTRAINT `umessages_message_sender_id_bcf82313_fk_auth_user_id` FOREIGN KEY (`sender_id`) REFERENCES `auth_user` (`id`) ON DELETE RESTRICT ON UPDATE RESTRICT)')

Would a more strict Exception handling suf·ice?

Reply:
check_permissions does not have a circular import issue, so this should be fixable too.

Reply:
@jmsmkn If you review/comment my two suggestions above, ill fix accordingly.